rest/client/methods.go

```diff
@@ -513,6 +513,29 @@ func (c *communicatorImpl) GetBannerMessage(ctx context.Context) (string, error)
 	return utility.FromStringPtr(banner.Text), nil
 }
 
+// GetEstimatedGeneratedTasks returns the estimated number of generated tasks to be created by an unfinalized patch.
+func (c *communicatorImpl) GetEstimatedGeneratedTasks(ctx context.Context, patchId string, tvPairs []serviceModel.TVPair) (int, error) {
+	if len(tvPairs) == 0 {
+		return 0, nil
+	}
+	info := requestInfo{
+		method: http.MethodGet,
+		path:   fmt.Sprintf("patches/%s/estimated_generated_tasks", patchId),
+	}
+	resp, err := c.request(ctx, info, tvPairs)
+	if err != nil {
+		return 0, errors.Wrap(err, "sending request to estimate number of activated generated tasks")
+	}
+	defer resp.Body.Close()
+
+	numTasksToFinalize := model.APINumTasksToFinalize{}
```

Comment:
This has to first check for non-OK statuses like [the other methods](https://github.com/evergreen-ci/evergreen/blob/35250e69a173022ad5d039affe7ea07ed83465fc/rest/client/methods.go#L1624-L1629), because if it errors, it could read the JSON (which is an error response) and interpret the nonexistent `num_tasks_to_finalize` field to mean it'll finalize 0 tasks.

(I think the admin-only communicator methods surrounding this one probably should also check the status is OK before assuming success but whatever...)